http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -514,7 +515,11 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift](
           config.persistentDiskMonitorConfig.create.maxAttempts,
           None
         )(
-          nowInstant.flatMap(now => persistentDiskQuery.updateSize(msg.diskId, msg.newSize, now).transaction[F]).void,
+          for {
+            _ <- nowInstant.flatMap(now => persistentDiskQuery.updateSize(msg.diskId, msg.newSize, now).transaction[F])
+            _ <- persistentDiskQuery.updateNeedsResize(msg.diskId, true).transaction[F]
+            _ <- msg.restartRuntimeId.traverse(restartRuntime(_))
```

Comment:
`restartRuntimeId` only exists if the `UpdateDiskMessage` came from a `updateRuntime` request. If it's coming from a `updateDisk` request, `restartRuntimeId` will be `None`. The reasoning behind this is because we have the `allowStop` field in `updateRuntime,` but it doesn't make sense for the `updateDisk` endpoint. So, if someone updates the disk size through the `updateRuntime` endpoint it will restart the runtime for the user if `allowStop` is true (or throw an error if `allowStop` is false). If someone updates the disk size through the `updateDisk` endpoint, the runtime will not be restarted but the disk will have `needsResize=true` until they restart the runtime themselves